import asyncio
from mds.agg_mds import datastore
from mds import config, logger
from urllib.parse import urlparse


# Okay, so what I need to do in this file is create a redis subscription
# Every time there is an upate on it, use it to update ES

if not config.USE_AGG_MDS:
    logger.info("aggregate MDS disabled")
    exit(1)

# 1. Init connections
# Setup connection to ES
# Might need to be worried about config?
# Not sure, but I know populate runs inside the pod right, so this should be okay
url_parts = urlparse(config.ES_ENDPOINT)
# gonna need to fix this await
async def try_this():
    await datastore.init(hostname=url_parts.hostname, port=url_parts.port)
    print(await datastore.get_status())
    await datastore.update_metadata("dumb_test", [{'1001_Genomes_Project': {'_guid_type': 'discovery_metadata', 'gen3_discovery': {'tags': [{'name': 'Aligned Reads', 'category': 'Condition'}], '_unique_id': '1000_Genomes_Project', 'study_description': 'The 1000 Genomes Project is a collaboration among research groups in the US, UK, and China and Germany to produce an extensive catalog of human genetic variation that will support future medical research studies. It will extend the data from the International HapMap Project, which created a resource that has been used to find more than 100 regions of the genome that are associated with common human diseases such as coronary artery disease and diabetes. The goal of the 1000 Genomes Project is to provide a resource of almost all variants, including SNPs and structural variants, and their haplotype contexts. This resource will allow genome-wide association studies to focus on almost all variants that exist in regions found to be associated with disease. The genomes of over 1000 unidentified individuals from around the world will be sequenced using next generation sequencing technologies. The results of the study will be publicly accessible to researchers worldwide.', 'full_name': '1000 Genomes Project', 'short_name': 'OpenAccess-1000_Genomes_Project', 'year': 'not_set', 'accession_number': '1000_Genomes_Project', 'commons': 'Gen3 Data Commons', 'study_url': 'https://www.genome.gov/27528684/1000-genomes-project', 'commons_name': 'Gen3'}}}, {'ACCOuNT_Clopidogrel_Arm': {'_guid_type': 'discovery_metadata', 'gen3_discovery': {'tags': [{'name': '', 'category': ''}], '_unique_id': 'ACCOuNT_Clopidogrel_Arm', 'study_description': None, 'full_name': 'Discovery, Clopidogrel Arm', 'short_name': 'not_set', 'year': 'not_set', 'accession_number': 'ACCOuNT_Clopidogrel_Arm', 'commons': 'Gen3 Data Commons', 'study_url': 'unknown', 'commons_name': 'Gen3'}}}, {'CCLE': {'_guid_type': 'discovery_metadata', 'gen3_discovery': {'tags': [{'name': 'Aligned Reads', 'category': 'Data Type'}], '_unique_id': 'CCLE', 'study_description': 'The CCLE (Cancer Cell Line Encyclopedia) project is a collaboration between the Broad Institute, and the Novartis Institutes for Biomedical Research and its Genomics Institute of the Novartis Research Foundation to conduct a detailed genetic and pharmacologic characterization of a large panel of human cancer models, to develop integrated computational analyses that link distinct pharmacologic vulnerabilities to genomic patterns and to translate cell line integrative genomics into cancer patient stratification. The CCLE provides public access to genomic data, analysis and visualization for over 1100 cell lines. The CCLE is an ongoing project and some data are not complete yet. The CCLE website is subject to periodic changes and improvements. Please visit regularly!', 'full_name': 'CCLE (Cancer Cell Line Encyclopedia)', 'short_name': 'OpenAccess-CCLE', 'year': 'not_set', 'accession_number': 'CCLE', 'commons': 'Gen3 Data Commons', 'study_url': 'https://portals.broadinstitute.org/ccle/about', 'commons_name': 'Gen3'}}}, {'ds000030': {'_guid_type': 'discovery_metadata', 'gen3_discovery': {'tags': [{'name': 'Imaging Files', 'category': 'Data Type'}], '_unique_id': 'ds000030', 'study_description': 'The Consortium for Neuropsychiatric Phenomics (CNP) is a large study funded by the NIH Roadmap Initiative that aims to facilitate discovery of the genetic and environmental bases of variation in psychological and neural system phenotypes, to elucidate the mechanisms that link the human genome to complex psychological syndromes, and to foster breakthroughs in the development of novel treatments for neuropsychiatric disorders. The study includes imaging of a large group of healthy individuals from the community (138 subjects), as well as samples of individuals diagnosed with schizoprenia (58), bipolar disorder (49), and ADHD (45). The participants, ages 21-50, were recruited by community advertisements from the Los Angeles area and completed extensive neuropsychogical testing, in addition to fMRI scanning. To be included individuals had to be either "White, Not of Hispanic or Latino Origin" or "Hispanic or Latino, of Any Race" following NIH designations of racial and ethnic minority groups, and have completed at least 8 years of education (other racial and ethnic minority groups were excluded because this was thought to increase risk of confounding planned genetic studies). For participants who spoke both English and Spanish, language for testing was determined by a verbal fluency test. Participants were screened for neurological disease, history of head injury with loss of consciousness or cognitive sequelae, use of psychoactive medications, substance dependence within past 6 months, history of major mental illness or ADHD, and current mood or anxiety disorder. Self-reported history of psychopathology was verified with the SCID-IV (First, Spitzer, Gibbon, & Williams, 1995). Urinalysis was used to screen for drugs of abuse (cannabis, amphetamine, opioids, cocaine, benzodiazepines) on the day of testing and excluded if results were positive.', 'full_name': 'UCLA Consortium for Neuropsychiatric Phenomics LA5c Study', 'short_name': 'OpenNeuro-ds000030', 'year': 'not_set', 'accession_number': 'ds000030', 'commons': 'Gen3 Data Commons', 'study_url': 'https://openneuro.org/datasets/ds000030/versions/00016', 'commons_name': 'Gen3'}}}, {'GSE63878': {'_guid_type': 'discovery_metadata', 'gen3_discovery': {'tags': [{'name': 'Array', 'category': 'Data Type'}], '_unique_id': 'GSE63878', 'study_description': 'The molecular factors involved in the development of Post-traumatic Stress Disorder (PTSD) remain poorly understood. Previous transcriptomic studies investigating the mechanisms of PTSD apply targeted approaches to identify individual genes under a cross-sectional framework lack a holistic view of the behaviours and properties of these genes at the system-level. Here we sought to apply an unsupervised gene-network-based approach to a prospective experimental design using whole-transcriptome RNA-Seq gene expression from peripheral blood leukocytes of U.S. Marines (N=188), obtained both pre- and post-deployment to conflict zones. We identified discrete groups of co-regulated genes (i.e., co-expression modules) and tested them for association to PTSD. We identified one module at both pre- and post-deployment containing putative causal signatures for PTSD development displaying an over-expression of genes enriched for functions of innate-immune response and interferon signalling (Type-I and Type-II). Importantly, these results were replicated in a second non-overlapping independent dataset of U.S. Marines (N=96), further outlining the role of innate immune and interferon signalling genes within co-expression modules to explain at least part of the causal pathophysiology for PTSD development. A second module, consequential of trauma exposure, contained PTSD resiliency signatures and an over-expression of genes involved in hemostasis and wound responsiveness suggesting that chronic levels of stress impair proper wound healing during/after exposure to the battlefield while highlighting the role of the hemostatic system as a clinical indicator of chronic-based stress. These findings provide novel insights for early preventative measures and advanced PTSD detection, which may lead to interventions that delay or perhaps abrogate the development of PTSD.\nWe used microarrays to characterize both prognostic and diagnostic molecular signatures associated to PTSD risk and PTSD status compared to control subjects.', 'full_name': 'Gene Networks Specific for Innate Immunity Define Post-traumatic Stress Disorder [Affymetrix]', 'short_name': 'GEO-GSE63878', 'year': 'not_set', 'accession_number': 'GSE63878', 'commons': 'Gen3 Data Commons', 'study_url': 'https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE63878', 'commons_name': 'Gen3'}}}], [], {}, {}, False)

asyncio.run(try_this())